### PR TITLE
Refactors github api call and finishes user story 3

### PR DESCRIPTION
**Pull request by rer7891**

### CODE REVIEW

File: app/facades/github_search.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  def display_repos
+    get_json('repos').take(5)
+  end
+
+  def display_following
+    get_json('following').take(5)
+  end
+
+  def find_github_resource(data_type, data)
```
Comment (rer7891) [inline app/facades/github_search.rb:14]: This takes the variable passed into the display methods i.e. display_repos and converts it into something we can instantiate a new object with such as Repo.new

File: app/facades/github_search.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    resource = data_type.singularize.capitalize
+    resource.constantize.new(data)
+  end
+
+  def get_json(data_type)
```
Comment (rer7891) [inline app/facades/github_search.rb:19]: This method can now be used for all display methods i.e. repos, following, followers. data_type is the variable from the display_repo/display_following methods

File: app/services/github_service.rb [hunk 3]
```diff
@@ -1,6 +1,11 @@
 class GithubService
-  def get_json_repos(token)
-    response = Faraday.get("https://api.github.com/user/repos?access_token=#{token}")
+  def get_json(url)
+    response = Faraday.get("https://api.github.com#{url}")
     JSON.parse(response.body, symbolize_names: true)
   end
+
```
Comment (rer7891) [inline app/services/github_service.rb:6]: This can now be used dynamically for all paths that start with /user

**Review by darren2802 — APPROVED** [review]
Looks good

**Pull request merged**